### ISSUE TITLE
Issue with modules named after stdlib modules

### ISSUE THREAD

**Issue by thisch** [PyCQA/pylint#3587]
pylint seems to incorrectly detect a self-import.

### Steps to reproduce
1. Create a file `/tmp/project/logging.py` with the following content

```
import logging
print(logging.CRITICAL)
```
2. Run pylint on this file `(cd /tmp && pylint project/logging.py)`

```
************* Module logging
project/logging.py:1:0: C0114: Missing module docstring (missing-module-docstring)
project/logging.py:1:0: W0406: Module import itself (import-self)
project/logging.py:4:6: E1101: Module 'logging' has no 'CRITICAL' member (no-member)
```

3. Run python `(cd /tmp && python project/logging.py)`
It outputs
`50`

### Expected behavior

Since python does not complain about a self-import, pylint should IMO also not complain about self-imports.

### pylint --version output

```
pylint 2.6.0-dev1
astroid 2.5.0
Python 3.8.2 | packaged by conda-forge | (default, Mar 23 2020, 18:16:37) 
[GCC 7.3.0]
```

**Comment by thisch** [comment]
It seems to make a difference when a `__init__.py` file is present:

`touch /tmp/project/__init__.py `

```
> pylint /tmp/project/logging.py                                                                                         
************* Module project.logging
/tmp/project/logging.py:1:0: C0114: Missing module docstring (missing-module-docstring)
```

Another issue (I think I have to create another ticket for it) is that if `--from-stdin` is used, the init file is not recognized.

```
cat /tmp/project/logging.py | pylint --from-stdin logging.py                                                           
************* Module logging
logging.py:1:0: C0114: Missing module docstring (missing-module-docstring)
logging.py:1:0: W0406: Module import itself (import-self)
logging.py:4:6: E1101: Module 'logging' has no 'CRITICAL' member (no-member)

----------------------------------------------------------------------
Your code has been rated at -25.00/10 (previous run: -25.00/10, +0.00)
```

**Comment by PCManticore** [comment]
Well the only issue I see here is that we should emit the error even when we have a `__init__` file, which currently does not happen. The code crashes when executed with Python, as in the following traceback, where we can see the module actually imports itself:
```
⋊> ~/p/p/p/3587 python project/logging.py                             08:52:19
<module 'logging' from '/Users/claudiu/projects/personal/playground/3587/project/logging.py'>
Traceback (most recent call last):
  File "project/logging.py", line 2, in <module>
    import logging
  File "/Users/claudiu/projects/personal/playground/3587/project/logging.py", line 3, in <module>
    print(logging.CRITICAL)
AttributeError: module 'logging' has no attribute 'CRITICAL'
```

**Comment by thisch** [comment]
Thx for having a look. 

I'm sry that point 3 in the description is wrong. Actually I only tested that the import works (`python -c 'import project.logging` and not that `python project/logging.py` works. I thought that the two calls are the same but apparently this is not the case.

Isn't there something similar to `python -c import project.logging` in `pylint`? I mean running `pylint on the complete folder or package must work (`pylint project`), but this does not seem to be the case.

**Comment by thisch** [comment]
**Here is my conclusion:**

Since `pylint project.logging` works (no `import-self` warning) and `python
project/logging.py` as well as `pylint project/logging.py` either fail with
an `AttributeError` or output a `import-self` warning, **I think we can close
this issue.**

`pylint` IMO can't do anything about properly handling self imports, since
`python` has the same problem with circular imports. I think this is an
expected behavior of cpython (correct me if I'm wrong).

I've seen several places on the web that suggest to pass all python files as
arguments to `pylint`. IMO this is not a good suggestion, since files like
the one in the description lead to warnings when passed as a filename
(instead of as a modulename) to pylint. It should be mentioned somewhere
in the docs how pylint should be run.

OT: I've just tried running `pylint project` (no `__init__.py`), because I
wanted to know whether it's possible to run pylint on a complete python-pkg,
but it fails with the following error:

```
************* Module project
project/__init__.py:1:0: F0010: error while code parsing: Unable to load file project/__init__.py:
[Errno 2] No such file or directory: 'project/__init__.py' (parse-error)

```

Let me know if I should create a ticket about this issue.

Note to myself: It is even possible to pass the file (logging.py) as stdin
to pylint, but it only succeeds (no `import-self` warning) if `project.logging` is
used as the name of the module:

``` cat project/logging.py| pylint --from-stdin project.logging ```

This means that projects like `flycheck` would have to somehow determine the
name of the module based on the filename. cc @cpitclaudel

**Comment by cpitclaudel** [comment]
> This means that projects like flycheck would have to somehow determine the name of the module based on the filename. cc @cpitclaudel

:fearful: my usual taje on this is: any logic that ends up in Flycheck to do this would be better-designed, better maintained, and less buggy if it's in pylint (there's no information we have that it doesn't have) :)

**Comment by thisch** [comment]
> > This means that projects like flycheck would have to somehow determine the name of the module based on the filename. cc @cpitclaudel
> 
>  my usual taje on this is: any logic that ends up in Flycheck to do this would be better-designed, better maintained, and less buggy if it's in pylint (there's no information we have that it doesn't have) :)

Don't be scared, since this issue IMO is only a minor one. Thx for your good suggestion, I think it totally makes sense to determine the module name in pylint itself. Unfortunately I don't think this is possible, since the existence of `__init__.py` files can no longer (since py3) be used to determine the module name of a file.

**Comment by thisch** [comment]
FYI I have a fix already for the `--from-stdin` feature for projects that have `__init__.py` files.

The fixes the original problem I had with the `self-import` in the `_pytest.logging` module.
I'll soon create a PR for it.
